lib/_stream_transform.js

```diff
@@ -115,9 +103,16 @@ function Transform(options) {
 
   Duplex.call(this, options);
 
-  this._transformState = new TransformState(this);
-
-  var stream = this;
+  this._transformState = {
+    afterTransform: (er, data) => {
+      return afterTransform(this, er, data);
+    },
```

Comment:
I think we should see if you can bring those improvements also to `_readableState` and `_writableState`, as allocating those are a hot path whenever using node streams.